com.unity.netcode.gameobjects/Runtime/NetworkVariable/NetworkVariableSerialization.cs

```diff
@@ -1255,6 +1289,8 @@ public static class NetworkVariableSerializationTypes
 #endif
         internal static void InitializeIntegerSerialization()
         {
+            NetworkVariableSerialization<byte>.Serializer = new ByteSerializer();
```

Comment:
This honestly shouldn't be needed either. There's an ILPP/codegen pass that should pick up the existence of `NetworkVariable<byte>` and generate correct initialization code to link it up to `UnmanagedTypeSerializer<byte>`. I'm 100% certain we have test code that verifies this as well. I'd be interested in knowing your use case to understand why this isn't happening for you.

Reply:
This fix is related to that ticket : 
https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/issues/2920
Beside that, I'm just trying to get going on easy low hanging fruits tickets and get used with current NGO workflow and I'm struggling with commit tags right now

Reply:
Having looked a little more at the code this morning, I understand why the codegen isn't picking this up - it's because a variable of a type other than `NetworkVariable<byte>` (such as `NetworkVariable<FixedString32Bytes>`) is serializing a `byte` value as part of its delta serialization. Delta serialization was added in NGO 1.9, which is why it doesn't happen in previous versions. This code assumes the existence of a `byte` serializer, but if the user doesn't have a `NetworkVariable<byte>` in their code, the `byte` serializer won't be initialized by the codegen pass.

There are actually two ways to approach this:

1) You can use the approach you started with here and initialize a `byte` serializer in this method. It just doesn't need its own serializer because there's no way to pack a byte, so it can (and should) use the generic `UnmanagedTypeSerializer<byte>`.

Additionally, you'll want to tell the codegen not to generate initialization for this since it's handled specially now. You can do that in `NetworkBehaviorILPP.cs` by adding `byte` to the `SpecialCaseTypes` list.

2) Alternatively (and I would say, more idiomatically, since this is how end users would do this): You could simply add the attribute `[GenerateSerializationForType(typeof(byte))]` to both the NetworkVariable and AnticipatedNetworkVariable classes. That'll do all the rest of the work for you.